src/Core/Workspace/Workspace.cs

```diff
@@ -1,557 +1,545 @@
 
-        /// <summary>
-        /// An event that is set after the workspace initialization has completed.
-        /// </summary>
-        private ManualResetEvent initialized = new ManualResetEvent(false);
+    /// <summary>
+    /// The compilation errors, if any.
+    /// </summary>
+    public IEnumerable<string> ErrorMessages { get; set; }
 
-        /// <inheritdoc/>
-        public Task Initialization => Task.Run(() => initialized.WaitOne());
+    /// <summary>
+    /// If any of the files in the workspace had any compilation errors.
+    /// </summary>
+    public bool HasErrors => ErrorMessages == null || ErrorMessages.Any();
 
-        /// <summary>
-        /// Main constructor that accepts ILogger and IReferences as dependencies.
-        /// </summary>
-        /// <param name="logger">Used to log messages to the console.</param>
-        /// <param name="references">List of references to use to compile the workspace.</param>
-        public Workspace(
-            IOptions<Settings> config,
-            ICompilerService compiler,
-            IReferences references,
-            INugetPackages packages,
-            ILogger<Workspace> logger,
-            IMetadataController metadata,
-            IEventService eventService)
+    /// <summary>
+    /// The folder where the project assemblies are permanently saved for cache.
+    /// </summary>
+    public string CacheFolder { get; set; }
+
+    /// <summary>
+    /// An event that is set after the workspace initialization has completed.
+    /// </summary>
+    private ManualResetEvent initialized = new ManualResetEvent(false);
+
+    /// <inheritdoc/>
+    public Task Initialization => Task.Run(() => initialized.WaitOne());
+
+    /// <summary>
+    /// Main constructor that accepts ILogger and IReferences as dependencies.
+    /// </summary>
+    /// <param name="logger">Used to log messages to the console.</param>
+    /// <param name="references">List of references to use to compile the workspace.</param>
+    public Workspace(
+        IOptions<Settings> config,
+        ICompilerService compiler,
+        IReferences references,
+        INugetPackages packages,
+        ILogger<Workspace> logger,
+        IMetadataController metadata,
+        IEventService eventService)
+    {
+        Compiler = compiler;
+        GlobalReferences = references;
+        NugetPackages = packages;
+        Logger = logger;
+
+        Root = config?.Value.Workspace;
+        CacheFolder = config?.Value.CacheFolder;
+        SkipAutoLoadProject = config?.Value.SkipAutoLoadProject ?? false;
+        MonitorWorkspace = config?.Value.MonitorWorkspace ?? false;
+
+        logger?.LogInformation(
+            "Starting IQ# Workspace:\n----------------\nRoot: {Root}\nCache folder:{CacheFolder}\nMonitoring changes: {MonitorWorkspace}\nUser agent: {UserAgent}\nHosting environment: {HostingEnvironment}\n----------------",
+            Root,
```

Comment:
What will be lost upon the following change?
```suggestion
        logger?.LogInformation(
            "Starting IQ# Workspace:\n" +
                "----------------\n" +
                "Root: {Root}\n" +
                "Cache folder:{CacheFolder}\n" +
                "Monitoring changes: {MonitorWorkspace}\n" +
                "User agent: {UserAgent}\n" +
                "Hosting environment: {HostingEnvironment}\n" +
                "----------------",
            Root,
```